Use port 80 for TLS origination by egress gateway

vadimeisenbergibm opened the pull request:
Resolves https://github.com/istio/istio/issues/9546.

The problem was that HTTP and HTTPS were mixed on port 443. The solution is to use port 80 of the gateway for HTTP with or without mTLS. (In case of HTTP/mTLS, specify the protocol as HTTPS).
The 443 port must be used for TLS only, with or without mTLS.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/3200#" title="Author self-approved">vadimeisenbergibm</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **frankbu**

If they are not already assigned, you can assign the PR to them by writing `/assign @frankbu` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["frankbu"]} -->

frankbu commented on content/docs/examples/advanced-gateways/egress-gateway-tls-origination/index.md:
```diff
@@ -86,8 +86,14 @@ be done by the egress gateway, as opposed to by the sidecar in the previous exam
+
+    {{< idea >}}
+    You may want to enable mutual TLS so the traffic between the source pod and the egress gateway will be encrypted.
+    In addition, mutual TLS will allow the egress gateway monitor the identity of the source pods and enable Mixer
```
to monitor

vadimeisenbergibm replied:
Fixed in https://github.com/istio/istio.io/pull/3200/commits/23f3d78a790334368dbf134daa7ab59ce0a08278.

frankbu commented:
@vadimeisenbergibm If I understand correctly, changing the doc like this will avoid the problem, but is there also some code changes needed to protect the gateway from crashing when configured the old way? That seems pretty nasty.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/3200#pullrequestreview-200669018" title="LGTM">frankbu</a>*, *<a href="https://github.com/istio/istio.io/pull/3200#" title="Author self-approved">vadimeisenbergibm</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[content/OWNERS](https://github.com/istio/istio.io/blob/master/content/OWNERS)~~ [frankbu]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

castlemilk commented:
@frankbu @vadimeisenbergibm, pilot doesn't crash, nor does the gateway. But as the issue describes in [stio/istio#9546](https://github.com/istio/istio/issues/9546), the egressgateway it enter a state that no longer allows egress traffic due to the config rejection. Effectively "crashing" or preventing all egress from that gateway. Not sure why this would be described as dramatic, it's a fairly severe defect if I can break egress simply by applying a gateway configuration which defines servers that specify `protocol: HTTPS` and `protocol: TLS` on the same port [443] (but different domains).

Please let me know if you aren't observing this behaviour.

vadimeisenbergibm commented:
@castlemilk I addressed your comment https://github.com/istio/istio.io/pull/3200#issuecomment-461199940 in https://github.com/istio/istio/issues/9546#issuecomment-461285270.

geeknoid commented:
If you rebase, the CI failure will go away.
